test/mocks/network/mocks.h

```diff
@@ -246,6 +247,46 @@ class MockListenSocket : public Socket {
   bool socket_is_open_ = true;
 };
 
+class MockSocket : public Socket {
```

Comment:
Sorry I should have mentioned this but we are trying to remove the monolothic mock files. Please add a new dedicated socket.h/cc file for the MockSocket.

Reply:
Sure. NP. :)